sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -55,6 +58,15 @@ public BufferHolder(UnsafeRow row, int initialSize) {
     this.buffer = new byte[fixedSize + initialSize];
     this.row = row;
     this.row.pointTo(buffer, buffer.length);
+    this.array = null;
+  }
+
+  public BufferHolder(UnsafeArrayData array, long numElements) {
```

Comment:
This is unnecessary. We are writing primitive arrays, so the size is fixed, we don't need to grow the buffer holder and re-point to it.